### PR TITLE
Enable go linter suite

### PR DESCRIPTION
**Pull request by troyronda**
- Upgrade to golangci-lint v1.18
- Enable gochecknoglobals, nolen and dupl
- Disallow naked returns
- Resolve warnings

Noted TODOs to review global usage.

Closes: #205
Closes: #216

Signed-off-by: Troy Ronda <troy@troyronda.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ linters-settings:
   maligned:
     suggest-new: true
   dupl:
-    threshold: 100
+    threshold: 150
```
Comment (troyronda) [inline .golangci.yml]: I reset to the default of 150.

File: .golangci.yml [hunk 2]
```diff
@@ -69,15 +69,21 @@ linters-settings:
       - opinionated
     disabled-checks:
       - unnamedResult
+  funlen:
+    lines: 60
```
Comment (troyronda) [inline .golangci.yml]: 60 is the default.

File: pkg/framework/aries/default.go [hunk 3]
```diff
@@ -21,9 +21,13 @@ import (
 	"github.com/hyperledger/aries-framework-go/pkg/storage/leveldb"
 )
 
-// DBPath Level DB Path.
-var dbPath = "/tmp/peerstore/"
-var defaultInboundPort = ":8090"
+// TODO handle the test scenario better (make dbPath constant).
+//nolint:gochecknoglobals
+var (
```
Comment (fqutishat) [inline pkg/framework/aries/default.go]: should be const

File: pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go [hunk 4]
```diff
@@ -26,9 +25,6 @@ const C20P = ContentEncryption("C20P") // Chacha20 encryption + Poly1035 authent
 // XC20P XChacha20Poly1035 algorithm
 const XC20P = ContentEncryption("XC20P") // XChacha20 encryption + Poly1035 authenticator cipher (192 bits nonce)
 
-// randReader is a cryptographically secure random number generator.
-var randReader = rand.Reader
```
Comment (Baha-sk) [inline pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go]: although not currently manipulated in tests, 
but this variable could be used to force random repeatable values in tests (inspired from go-jose).

Comment (troyronda) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go]: These things should be documented / commented. Global variables raises flags.

Comment (troyronda) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go]: Also - we shouldn't do these things until we actually are using them :).

File: .golangci.yml [hunk 5]
```diff
@@ -69,15 +69,24 @@ linters-settings:
...
   enable-all: true
   disable:
     - maligned
     - prealloc
-    - dupl
-    - gochecknoglobals
 
 issues:
   exclude-use-default: false
+  exclude-rules:
```
Comment (troyronda) [inline .golangci.yml]: Any else that should be excluded from tests?

File: pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go (hunk 4 above)
Comment (Baha-sk) [reply pkg/didcomm/crypto/jwe/authcrypt/authcrypt.go]: sure.. it can be removed then

**Pull request merged**